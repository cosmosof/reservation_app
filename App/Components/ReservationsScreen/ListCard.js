// @flow

import * as React from 'react';
import {Text, View} from 'react-native';
import styles from './Styles/ListCardStyles';

type Props = { data: Object, styles: Object };

export default function ListCard(props: Props) {
  const {container, value, label, oddRow, row} = styles;
  const {name, id, hotelName, arrivalDate, departureDate} = props.data;
  return (
    <View
      key={id}
      style={container}
    >
      <View style={oddRow}>
        <Text style={label}>Guest name </Text>
        <Text style={value}> {name} </Text>
      </View>
      <View style={row}>
        <Text style={label}>Hotel name </Text>
        <Text style={value}> {hotelName} </Text>
      </View>
      <View style={oddRow}>
        <Text style={label}>Arrival date</Text>
        <Text style={value}> {arrivalDate} </Text>
      </View>
      <View style={row}>
        <Text style={label}>Departure date</Text>
        <Text style={value}> {departureDate} </Text>
      </View>
    </View>
  );
}
