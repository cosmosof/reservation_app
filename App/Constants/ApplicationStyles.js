import {Platform} from 'react-native';
import Fonts from './Fonts';
import Colors from './Colors';

const ApplicationStyles = {
  navbarHeader: {
    color: Colors.darkMatBlue2,
    fontSize: 18,
    fontWeight: '600',
    fontFamily: Platform.OS == 'ios' ? Fonts.typeIOS.condensed : Fonts.typeAndroid.condensed,
  },
};

export default {
  ApplicationStyles,
};
