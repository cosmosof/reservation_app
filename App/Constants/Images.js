const images = {
  logo: require('../Images/Relate.png'),
};

export default images;
