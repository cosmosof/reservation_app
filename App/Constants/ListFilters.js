export default [
  {
    index: 0,
    name: 'guest name',
    value: 'name',
  },
  {
    index: 1,
    name: 'in guest names',
    value: 'name_contains',
  },
  {
    index: 3,
    name: 'name starts with',
    value: 'name_starts_with',
  },
  {
    index: 4,
    name: 'hotel name',
    value: 'hotelName',
  },
  {
    index: 5,
    name: 'in hotel names',
    value: 'hotelName_contains',
  },
  {
    index: 7,
    name: 'hotel names start with',
    value: 'hotelName_starts_with',
  },
  {
    index: 9,
    name: 'arrival date',
    value: 'arrivalDate',
  },
  {
    index: 10,
    name: 'departure date',
    value: 'departureDate',
  },
];
