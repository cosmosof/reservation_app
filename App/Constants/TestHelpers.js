const noop = () => {};
const sampleText = 'Sample text';
export default {
  noop,
  sampleText,
};
