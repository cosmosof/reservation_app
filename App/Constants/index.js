import Colors from './Colors';
import Fonts from './Fonts';
import Metrics from './Metrics';
import Images from './Images';
import ListFilters from './ListFilters';
import ApplicationStyles from './ApplicationStyles';
import TestHelpers from './TestHelpers';

export {Colors, Fonts, Images, Metrics, ListFilters, ApplicationStyles, TestHelpers};
