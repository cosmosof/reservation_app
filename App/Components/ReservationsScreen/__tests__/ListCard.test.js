import React from 'react';
import {shallow} from 'enzyme';
import ListCard from '../ListCard';
import {TestHelpers} from '../../../Constants';

const {sampleText} = TestHelpers;

describe('<ListCard>', () => {
  const Component = <ListCard />;
  describe('Structure', () => {
    it('renders correctly if data props is set', () => {
      const Clone = React.cloneElement(Component, {data: {name: sampleText}});
      const wrapper = shallow(Clone);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
