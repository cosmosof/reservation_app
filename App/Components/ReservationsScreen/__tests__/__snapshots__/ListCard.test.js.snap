// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListCard> Structure renders correctly if data props is set 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderBottomWidth": 0.5,
      "borderColor": "#CCCCCC",
      "borderLeftWidth": 0.5,
      "borderRightWidth": 0.5,
      "borderTopWidth": 0.5,
      "borderWidth": 1,
      "flex": 1,
      "flexDirection": "column",
      "marginBottom": 5,
      "marginLeft": 5,
      "marginRight": 5,
      "padding": 30,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#6E7C81",
          "fontFamily": "AvenirNext-Regular",
          "fontSize": 14,
          "fontWeight": "400",
        }
      }
    >
      Guest name 
    </Text>
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#6E7C81",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
       
      Sample text
       
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F2FBFF",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#6E7C81",
          "fontFamily": "AvenirNext-Regular",
          "fontSize": 14,
          "fontWeight": "400",
        }
      }
    >
      Hotel name 
    </Text>
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#6E7C81",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
       
       
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#6E7C81",
          "fontFamily": "AvenirNext-Regular",
          "fontSize": 14,
          "fontWeight": "400",
        }
      }
    >
      Arrival date
    </Text>
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#6E7C81",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
       
       
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#F2FBFF",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "flex-start",
          "color": "#6E7C81",
          "fontFamily": "AvenirNext-Regular",
          "fontSize": 14,
          "fontWeight": "400",
        }
      }
    >
      Departure date
    </Text>
    <Text
      style={
        Object {
          "alignItems": "flex-end",
          "color": "#6E7C81",
          "fontFamily": "AvenirNextCondensed-DemiBold",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
       
       
    </Text>
  </View>
</View>
`;
